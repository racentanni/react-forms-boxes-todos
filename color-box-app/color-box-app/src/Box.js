import React from 'react';

function Box({ id, width, height, backgroundColor, removeBox }) {
  const handleRemove = () => removeBox(id);

  return (
    <div className='Box'>
      <div 
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor: backgroundColor
        }}
      ></div>
      <button onClick={handleRemove}>X</button>
    </div>
  );
}

export default Box;