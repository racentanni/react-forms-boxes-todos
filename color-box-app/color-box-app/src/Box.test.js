import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', () => {
  render(<Box id="1" width={100} height={100} backgroundColor="red" removeBox={() => {}} />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<Box id="1" width={100} height={100} backgroundColor="red" removeBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

it('calls removeBox on button click', () => {
  const removeBox = jest.fn();
  const { getByText } = render(<Box id="1" width={100} height={100} backgroundColor="red" removeBox={removeBox} />);
  fireEvent.click(getByText('X'));
  expect(removeBox).toHaveBeenCalledWith("1");
});