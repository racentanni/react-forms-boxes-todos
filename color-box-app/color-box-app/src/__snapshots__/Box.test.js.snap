// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="width: 100px; height: 100px; background-color: red;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
